pkg/cmd/inspect.go

```diff
@@ -155,6 +155,93 @@ func (o *InspectOptions) Validate() error {
 	return nil
 }
 
+func objectReferenceToResourceInfo(clientGetter genericclioptions.RESTClientGetter, ref *configv1.ObjectReference) (*resource.Info, error) {
+	b := resource.NewBuilder(clientGetter).
+		Unstructured().
+		ResourceTypeOrNameArgs(false, fmt.Sprintf("%s/%s", ref.Resource, ref.Name)).
+		NamespaceParam(ref.Namespace).
+		Flatten().
+		Latest()
+
+	infos, err := b.Do().Infos()
+	if err != nil {
+		return nil, err
+	}
+
+	return infos[0], nil
+}
+
+func infoToContextKey(info *resource.Info) string {
```

Comment:
have a variant `objectRefToContextKey` so you don't have to retrieve things you've already seen.